fdbserver/KeyValueStoreRocksDB.actor.cpp

```diff
@@ -383,31 +470,35 @@ struct RocksDBKeyValueStore : IKeyValueStore {
 			Optional<UID> debugID;
 			double startTime;
 			ThreadReturnPromise<Optional<Value>> result;
-			ReadValueAction(KeyRef key, Optional<UID> debugID)
-			  : key(key), debugID(debugID), startTime(timer_monotonic()) {}
+			ReadValueAction(KeyRef key, Optional<UID> debugID) : key(key), debugID(debugID), startTime(now()) {}
 			double getTimeEstimate() const override { return SERVER_KNOBS->READ_VALUE_TIME_ESTIMATE; }
 		};
 		void action(ReadValueAction& a) {
+			readValueQueueWaitHistogram->sampleSeconds(now() - a.startTime);
 			Optional<TraceBatch> traceBatch;
 			if (a.debugID.present()) {
 				traceBatch = { TraceBatch{} };
 				traceBatch.get().addEvent("GetValueDebug", a.debugID.get().first(), "Reader.Before");
 			}
-			if (timer_monotonic() - a.startTime > readValueTimeout) {
+			if (now() - a.startTime > readValueTimeout) {
 				TraceEvent(SevWarn, "RocksDBError")
 				    .detail("Error", "Read value request timedout")
 				    .detail("Method", "ReadValueAction")
 				    .detail("Timeout value", readValueTimeout);
 				a.result.sendError(transaction_too_old());
 				return;
 			}
+			double readBeginTime = now();
 			rocksdb::PinnableSlice value;
 			auto options = getReadOptions();
-			uint64_t deadlineMircos =
-			    db->GetEnv()->NowMicros() + (readValueTimeout - (timer_monotonic() - a.startTime)) * 1000000;
+			uint64_t deadlineMircos = db->GetEnv()->NowMicros() + (readValueTimeout - (now() - a.startTime)) * 1000000;
```

Comment:
why changing to now()?

Reply:
+1

Reply:
As we discussed, we will be using timer() in simulation and timer_monotonic in non-simulation. To reduce the cost of timer_monotonic, I am sampling at the rate of approx 1 out of 1000 read ops.